tests/cli.rs

```diff
@@ -0,0 +1,169 @@
+use std::io::{Read, Write};
+use std::iter::repeat;
+use std::process::{Command, Stdio};
+
+fn run_script<T: AsRef<str>>(commands: &[T]) -> Vec<String> {
+    let mut child = Command::new("cargo")
+        .arg("run")
+        .stdin(Stdio::piped())
+        .stdout(Stdio::piped())
+        .spawn()
+        .expect("Failed to spawn child process");
+
+    let mut stdin = child.stdin.take().expect("Failed to open stdin");
+    let stdout = &mut child.stdout.take().expect("Failed to open stdout");
+    let cmds = commands.iter();
+    let mut outputs = vec![];
```

Comment:
I am going to remove this special bytes processing portion I think in the next iteration. This turned out to be useless I think. I will check.